### ISSUE TITLE
Website: Central Reservation System (CRS) - Booking API

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#501]
## 🚀 Feature Summary

The Central Reservation System (CRS) acts as the core booking engine for all properties and channels. It centralizes reservations, availability, and pricing while ensuring real-time synchronization across distribution partners. The CRS enables accurate inventory control, seamless booking flows, and scalable multi-property operations.

----

[PMS](https://github.com/domits1/Domits/issues/149) Capabilities: [Listings](https://github.com/domits1/Domits/issues/2792), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [Direct Bookings](https://github.com/domits1/Domits/issues/2600), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Tasks](https://github.com/domits1/Domits/issues/2794), [Finance](https://github.com/domits1/Domits/issues/163), [Reviews](https://github.com/domits1/Domits/issues/90), [Intelligence](https://github.com/domits1/Domits/issues/1490), [Settings](https://github.com/domits1/Domits/issues/2793), [Integrations](https://github.com/domits1/Domits/issues/433), [Agentic AI](https://github.com/domits1/Domits/issues/442)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [Listings](https://github.com/domits1/Domits/issues/3039), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [Direct Bookings](https://github.com/domits1/Domits/issues/3038), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Tasks](https://github.com/domits1/Domits/issues/3040), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/a36519e2-eee8-44ec-9504-c5ef5cdd2c88" />

##  **Reservation management**

**General**
- [x] Database table booking
- [x] UX/UI Design

**Reservations CRUD & Statuses**
- [ ] #1746 
- [x] #1747
- [ ] #1646
- [x] #1996
- [ ] #1750

**Payments**
- [x] #1982
- [x] #2058 

**Reservations Dashboards**
- [ ] #2257 
- [x] #3076 
- [ ] #512 
- [x] #1997
- [ ] #3133 

**Reservations Reporting**
- [ ] #1751
- [x] #1777 
- [ ] #1773
- [ ] #1775
- [ ] #1776

**Next phase**
- [ ] #1574
- [ ] #1586 
- [ ] #1645
- [ ] #1647
- [ ] #1649
- [ ] #1651
- [ ] #1652
- [ ] #1653
- [ ] #1655
- [ ] #1656
- [ ] #1657
- [ ] #1658 
- [x] #1672
- [ ] #1680
- [ ] #1752 
- [ ] #3136 

Steps to take:
- [ ] Add implementation of "reserved" status for Booking records
- [ ] Add expiry dates for Bookings when they are reserved
- [ ] New API endpoint and/or lambda functions for adding Bookings without paying for them instantly

<img width="641" alt="Schermafbeelding 2025-01-10 om 16 19 38" src="https://github.com/user-attachments/assets/1b5b884d-f81b-4d0e-a856-8970ae7cd331" />

<img width="495" alt="Schermafbeelding 2025-01-10 om 16 20 57" src="https://github.com/user-attachments/assets/2fa1a3b6-efb3-475c-b31b-efc9368d3a94" />

Resources:
https://www.altexsoft.com/blog/central-reservation-system-hotel/

V3 Database: https://dbdiagram.io/d/DynamoDB-Property-Table-ERD-67c5ce64263d6cf9a00f4c0a

Wiki: https://github.com/domits1/Domits/wiki/Web-reservations-with-delayed-paying

---

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No Central Reservation System

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Central Reservation System

---

## 👥 Target Users
Who benefits from this feature?
- [ ] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

- [ ] Given X, when Y, then Z

### Step 1: Let Hosts Set Payment Deadlines
- Find the function that adds accommodations to the database.
- Update it so hosts can choose:
  - How many hours after booking guests must pay (e.g., 12, 24, 48 hours).
  - How many hours before check-in payment must be made (e.g., at least 3 days before check-in).
- If the host doesn’t set anything, we’ll use defaults:
  - 24 hours after booking.
  - 48 hours before check-in.
- Save this info when a new accommodation is created.

### Step 2: Adjust Booking Creation to Use Expiry Rules
- Update the function that creates a reservation.
- Get the payment deadline from the accommodation data.
- Decide when the reservation expires:
  - X hours after booking **OR**
  - X hours before check-in (**whichever comes first**).
- Store this expiration time in the reservation and mark it as **"pending"** instead of making the guest pay instantly.

### Step 3: Let Guests Pay Later
- Guests can see their pending reservation and pay when they’re ready.
- Update the payment function:
  - Find the host’s Stripe account.
  - Only allow payments for **pending** reservations.
  - If payment is successful, update the reservation to **"confirmed"**.

### Step 4: Auto-Cancel Unpaid Reservations
- Modify the function that cancels unpaid reservations.
- Check which reservations have expired.
- If they’re still unpaid, mark them as **"expired"**.
- Run this check every hour.

### Step 5: Update the UI
- Let hosts choose their payment rules when they list a property.
- Show guests the payment deadline when they book.
- Add a **"Pay Now"** button to the guest dashboard.
A central database for room inventory, simplifying management of all types of reservations including individual, group, and party, company, travel agent, multilegged, multi-rate, and waitlisted.

### 1. Handling Edge Cases
- Define how last-minute bookings close to check-in should be handled.
- Consider requiring instant payment for bookings with a certain time frame before check-in.

### 2. Default Settings & Overrides
- Clarify if hosts can disable payment deadlines completely, requiring full payment upfront.

### 3. Notification System
- Implement email, SMS or push (app) reminders for upcoming payment deadlines.
- Also notify guests when a reservation is about to expire

### 4. Admin Controls & Overrides
- Determine if admins can modify payment deadlines after booking
- Define if hosts can manually override expiration rules for special cases.

### 5. Concurrency Handling
- Prevent conflicts if a guest attempts to pay at the exact moment their reservation expires.
- Consider adding a short time window after a reservation expires. This helps to prevent cases where a guest attempts to pay just as their reservation expires, which could reduce frustation and potential support issues.

**Comment by Bambaclad1** [comment]
Perhaps a nice issue template to document the issues in:

### Description
Describe the issue here, what it is about, and anything else.

### User Story
"As a host, / As a guest,
I want a ... experience to ... for a management with ..."

### Acceptance Criteria

- [ ] Acceptance Criteria 1
- [ ] Acceptance Criteria 2
- [ ] Acceptance Criteria 3
- [ ] ...


### Related Issues
This issue is related to #... 

### Additional Notes
- Ensure consistency...
- ...
